src/deprecates.jl

```diff
@@ -20,6 +20,15 @@ import Base.varm, Base.stdm
 @deprecate AICc(obj::StatisticalModel) aicc(obj)
 @deprecate BIC(obj::StatisticalModel) bic(obj)
 
+if !isdefined(Base, :stderr)
+    @deprecate stderr(obj::StatisticalModel) stderror(obj)
+else
+    function (io::typeof(stderr))(obj::StatisticalModel)
```

Comment:
The problem with this is that `typeof(stderr)` can change at runtime if `redirect_stderr` is called.

So this should just be `::IO` to catch all usages. I don't think it is a problem to have a fairly general method here as a temporary deprecation since it continues to throw a MethodError for non-`stderr` objects. (People are unlikely to pass `StatisticalModel` objects to random IO objects by accident anyway.)

Reply:
Why doesn't `IO` work?  On 0.7 (unless something has changed very recently), `typeof(stderr)` is `TTY`, which is a subtype of `IO`.

Reply:
> I'm not sure we should care about supporting that deprecation after calling `redirect_stderr`

This would mean that the deprecation wouldn't work in `IJulia`, for example.

Reply:
Because of https://github.com/JuliaLang/julia/issues/14919:
```julia
julia> (io::IO)() = 1
ERROR: cannot add methods to an abstract type
```

Reply:
I see… yes, I can't think of a way to make it work for IJulia (without adding an IJulia dependency), since IJulia redirects to [its own `IO` type](https://github.com/JuliaLang/IJulia.jl/blob/9cba26457abe25058e1331b76f7b117406d72507/src/stdio.jl#L7-L9).